dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ProviderConfig.java

```diff
@@ -157,7 +158,7 @@
 
     @Deprecated
     public void setProtocol(String protocol) {
-        this.protocols = Arrays.asList(new ProtocolConfig[]{new ProtocolConfig(protocol)});
+        this.protocols = new ArrayList<ProtocolConfig>(Arrays.asList(new ProtocolConfig[]{ new ProtocolConfig(protocol)}));

```

Comment:
`this.protocols = new ArrayList<>(new ProtocolConfig(protocol));` seems better to me

Reply:
protocol is a string, we need to build a ProtocolConfig object to pass to ArrayList class, I believe this is just the way to use. Even though it seems a little bit hard to read :(